### PR TITLE
SSCS-5280 Add functional tests for DL6

### PR DESCRIPTION
**Pull request by jackmaloney**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-5280

### Change description ###
Create the functional tests for the DL6 template

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/EvidenceShareFunctionalTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        String json = getJson(SYA_APPEAL_CREATED);
+        json = json.replace("CASE_ID_TO_BE_REPLACED", ccdCaseId);
+
+        simulateCcdCallback(json);
+
+        SscsCaseDetails caseDetails = findCaseById(ccdCaseId);
+
+        List<SscsDocument> docs = caseDetails.getData().getSscsDocument();
+        assertEquals(1, docs.size());
+        assertEquals("DL6-" + ccdCaseId + ".pdf", docs.get(0).getValue().getDocumentFileName());
```
Comment (ryandac) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/EvidenceShareFunctionalTest.java]: great test 👍

File: src/main/java/uk/gov/hmcts/reform/sscs/service/DocumentPlaceholderService.java [hunk 2]
```diff
@@ -27,11 +26,11 @@ public DocumentPlaceholderService(DwpAddressLookup dwpAddressLookup) {
         this.dwpAddressLookup = dwpAddressLookup;
     }
 
-    public Map<String, Object> generatePlaceholders(SscsCaseData caseData) {
+    public Map<String, Object> generatePlaceholders(SscsCaseData caseData, LocalDateTime caseCreatedDate) {
         Map<String, Object> placeholders = new HashMap<>();
 
         Appeal appeal = caseData.getAppeal();
-        placeholders.put(CASE_CREATED_DATE_LITERAL, formatDate(caseData.getCaseCreated()));
+        placeholders.put(CASE_CREATED_DATE_LITERAL, caseCreatedDate.toLocalDate().toString());
```
[the remark this answers is not in the document]
Comment (jackmaloney) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/DocumentPlaceholderService.java]: Na

File: src/test/java/uk/gov/hmcts/reform/sscs/service/DocumentPlaceholderServiceTest.java [hunk 3]
```diff
@@ -77,9 +80,9 @@ public void givenACaseDataWithNoRpc_thenGenerateThePlaceholderMappingsWithoutRpc
         when(dwpAddressLookup.lookup("PIP", "1"))
             .thenReturn(Optional.of(new DwpAddress(RPC_ADDRESS1, RPC_ADDRESS2, RPC_CITY, POSTCODE)));
 
-        Map<String, Object> result = service.generatePlaceholders(caseData);
+        Map<String, Object> result = service.generatePlaceholders(caseData, now);
 
-        assertEquals(result.get(CASE_CREATED_DATE_LITERAL), "2018-10-23");
+        assertEquals(now.toLocalDate().toString(), result.get(CASE_CREATED_DATE_LITERAL));
```
[the remark this answers is not in the document]
Comment (jackmaloney) [reply src/test/java/uk/gov/hmcts/reform/sscs/service/DocumentPlaceholderServiceTest.java]: Na

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/EvidenceShareFunctionalTest.java (hunk 1 above)
Comment (jackmaloney) [reply src/e2e/java/uk/gov/hmcts/reform/sscs/functional/EvidenceShareFunctionalTest.java]: thanks

**Pull request merged**